### PR TITLE
[Al Amin Step-17] login & logout implementation

### PR DESCRIPTION
**Pull request by rr-mdalauddin-alamin**
- Let's implement it ourselves without using additional gems
  - By not using Gem such as Devise, the purpose is to deepen the understanding of the mechanism such as HTTP cookies and Session in Rails.
  - It also aims to deepen your understanding of general authentication (such as password handling).
- Let's implement a login screen
- If you are not logged in, let's prevent you from transitioning to the task management page
- Display the tasks only you created
- Let's implement the logout function

### CODE REVIEW

File: myapp/app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -1,10 +1,11 @@
 # frozen_string_literal: true
 
 class TasksController < ApplicationController # rubocop:todo Style/Documentation
+  before_action :check_authentication
   def index
     @query = Task.ransack(params[:q])
-
-    @tasks = @query.result.custom_order(params[:sort]).page(params[:page])
+    puts(@current_user["id"])
```
Comment (mt-ytaher) [inline myapp/app/controllers/tasks_controller.rb]: can't we use `current_user['id']` here? As you already created current_user method in the application controller

Note: I didn't run the code, it just my assumption

Comment (rr-mdalauddin-alamin) [reply myapp/app/controllers/tasks_controller.rb]: Yeah we can, actually i was checking the value and end up forgetting to remove before committing. 🤣

Comment (rr-mdalauddin-alamin) [reply myapp/app/controllers/tasks_controller.rb]: removed

File: myapp/app/controllers/tasks_controller.rb [hunk 2]
```diff
@@ -23,7 +23,7 @@ def edit
 
   def create
     @task = Task.new(task_params)
-    @task.user_id = 1 # because user must exist
+    @task.user_id = current_user['id']
     @task.user_type = 'Task' # user_type is needed because of NOT NULL
```
Comment (mt-ytaher) [inline myapp/app/controllers/tasks_controller.rb:27]: [nitpick] Not related to this. But why is `user_type` a task? Shouldn't `user_type` be like normal, admin, or something like that?

**Review by mt-ytaher — APPROVED** [review]
LGTM

File: myapp/app/controllers/tasks_controller.rb (hunk 2 above)
Comment (rr-mdalauddin-alamin) [reply myapp/app/controllers/tasks_controller.rb:27]: that was generated automatically due to use `polymorphic: true`. I'll migrate it.

**Pull request merged**